### PR TITLE
[action] remove config.EVMNetworkID() from action and api package

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
global `config.EVMNetworkID()` is a work-around, the better solution is to pass an `EvmNetworkID` at creation time, or pass it via context

## Type of change
- [x] Code refactor or improvement

# How Has This Been Tested?
- [x] make test

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: action/action_deserializer.go
Comment (dustinxie) [inline action/action_deserializer.go:17]: explanatory comments per haixiang's earlier comment

File: blockchain/block/blockstore.go
Comment (dustinxie) [inline blockchain/block/blockstore.go:46]: will file task for team member to work on

File: blockchain/filedao/filedao_legacy.go
Comment (dustinxie) [inline blockchain/filedao/filedao_legacy.go:267]: after PR merge, will file task for team member to work on

File: consensus/scheme/rolldpos/blockproposal.go
Comment (dustinxie) [inline consensus/scheme/rolldpos/blockproposal.go:67]: after PR merge, will file task for team member to work on

File: consensus/scheme/rolldpos/endorsementmanager.go
Comment (dustinxie) [inline consensus/scheme/rolldpos/endorsementmanager.go:130]: after PR merge, will file task for team member to work on

File: action/action_deserializer.go
Comment (dustinxie) [reply action/action_deserializer.go:17]: a possibly better solution might be:
```
act, err := action.NewSerializer(ctx).ActionToSealedEnvelope() // NewSerializer will extract evmNetworkID from ctx
```
so we don't need to explicitly call `SetEvmNetworkID(id)` everywhere

however this is impossible now due to import cycle, moving `context.go` from `/action/protocol` to `/action` could unblock this, maybe a further task

File: action/sealedenvelope.go
Comment (Liuhaai) [inline action/sealedenvelope.go:94]: evmID

File: action/action_deserializer.go
Comment (Liuhaai) [inline action/action_deserializer.go:31]: Since Deserializer has to be used, `selp.LoadProto` can be changed to private func?

File: blockchain/block/block_deserializer.go
Comment (Liuhaai) [inline blockchain/block/block_deserializer.go:39]: Difference between `Deserializer.FromBlockProto` and `Block.ConvertFromBlockPb`?

Comment (dustinxie) [reply blockchain/block/block_deserializer.go:39]: they are the same, looks like `Block.ConvertFromBlockPb` can be removed (only test code used it), will file a new PR to do this (and change `selp.LoadProto` to private func)

File: action/sealedenvelope.go
Comment (dustinxie) [reply action/sealedenvelope.go]: fixed

File: action/action_deserializer.go
Comment (dustinxie) [reply action/action_deserializer.go:31]: will file a new PR to do this

**Comment by Liuhaai** [comment]
could block Deserializer be combined together with action Deserializer?

**Comment by dustinxie** [comment]
> could block Deserializer be combined together with action Deserializer?

programming-wise yes, but I think keeping them separate is clear (and easy for understanding)? you have code like `action.Deserializer{}.xxx` and `block.Deserializer{}.xxx`, which contributes to better readability

**Comment by Liuhaai** [comment]
> > could block Deserializer be combined together with action Deserializer?
> 
> programming-wise yes, but I think keeping them separate is clear (and easy for understanding)? you have code like `action.Deserializer{}.xxx` and `block.Deserializer{}.xxx`, which contributes to better readability

Good. Thx for explanation.

File: blockchain/block/body.go
Comment (Liuhaai) [inline blockchain/block/body.go:41]: With blockDeserializer module, body.LoadProto() and body.Deserializer() could be removed?

File: blockchain/block/block_deserializer.go
Comment (Liuhaai) [reply blockchain/block/block_deserializer.go:39]: The change is not large. Could you finish it in this pr?

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**
